spec/models/order_spec.rb

```diff
@@ -0,0 +1,70 @@
+require "rails_helper"
+
+RSpec.describe Order, type: :model do
+  describe "Associations" do
+    it {
+      should belong_to(:user)
+      should belong_to(:discount).optional(:true)
+      should belong_to(:address).optional(:true)
+      should have_many(:order_details).dependent(:destroy)
+      should have_many(:book_details).through(:order_details)
+    }
+  end
+
+  describe "Enums" do
+    it {
+      should define_enum_for(:status)
+        .with_values(pending: 0, accepted: 1, rejected: 2)
+        .with_suffix
+    }
+  end
+
+  describe "#send_mail_approve" do
+    let!(:order){FactoryBot.create(:order)}
```

Comment:
phân biệt giữa let và let!? ở đây dùng let đc ko em

Reply:
let được gọi khi khối it đầu tiên trong example group được thực thi, những lần sau thì sẽ lấy giá trị được cached lại, còn let! được gọi ở trước mọi khối it trong example group. Ở trường hợp test này của em thì mỗi order chỉ có 1 lần nhận mail nên em nghĩ dùng let!

Reply:
Dùng let thôi em nhé, chỉ trường hợp bắt buộc mới dùng let!